lib/parser.js

```diff
@@ -1,37 +1,24 @@
 'use strict';
 
-// glorious streaming json parser, built specifically for the twitter streaming api
-// assumptions:
-//   1) ninjas are mammals
-//   2) tweets come in chunks of text, surrounded by {}'s, separated by line breaks
-//   3) only one tweet per chunk
-//
-//   p = new parser.instance()
-//   p.addListener('object', function...)
-//   p.receive(data)
-//   p.receive(data)
-//   ...
-
-var EventEmitter = require('events').EventEmitter;
+var Transform = require('stream').Transform;
 
 var Parser = module.exports = function Parser() {
-  // Make sure we call our parents constructor
-  EventEmitter.call(this);
+  Transform.call(this, {
+    objectMode: true
```

Comment:
Per https://nodejs.org/api/stream.html#stream_new_stream_duplex_options setting only `writableObjectMode` would be more appropriate as the input (readable mode) is still text-based